File: py-scoring/src/ai/h2o/sparkling/ml/models/H2OMOJOSettings.py
from pyspark.ml.param import *
from pyspark.ml.wrapper import JavaWrapper


class H2OMOJOSettings(JavaWrapper):

    def __init__(self,
                 predictionCol="prediction",
                 detailedPredictionCol="detailed_prediction",
                 convertUnknownCategoricalLevelsToNa=False,
                 convertInvalidNumbersToNa=False,
                 withContributions=False,
                 withInternalContributions=False,
                 withPredictionInterval=False,
                 withLeafNodeAssignments=False,
                 withStageResults=False,
                 dataFrameSerializer="ai.h2o.sparkling.utils.JSONDataFrameSerializer",
                 scoringBulkSize=1000):
        self._java_obj = None

        assert isinstance(predictionCol, str)
        assert isinstance(detailedPredictionCol, str)
        assert isinstance(convertUnknownCategoricalLevelsToNa, bool)
        assert isinstance(convertInvalidNumbersToNa, bool)
        assert isinstance(withContributions, bool)
        assert isinstance(withInternalContributions, bool)
        assert isinstance(withPredictionInterval, bool)
        assert isinstance(withLeafNodeAssignments, bool)
        assert isinstance(withStageResults, bool)
        assert isinstance(dataFrameSerializer, str)
        assert isinstance(scoringBulkSize, int)
        self.predictionCol = predictionCol
        self.detailedPredictionCol = detailedPredictionCol
        self.convertUnknownCategoricalLevelsToNa = convertUnknownCategoricalLevelsToNa
        self.convertInvalidNumbersToNa = convertInvalidNumbersToNa
        self.withContributions = withContributions
        self.withInternalContributions = withInternalContributions
        self.withPredictionInterval = withPredictionInterval
        self.withLeafNodeAssignments = withLeafNodeAssignments
        self.withStageResults = withStageResults
        self.dataFrameSerializer = dataFrameSerializer
        self.scoringBulkSize = scoringBulkSize

    def toJavaObject(self):
        self._java_obj = self._new_java_obj("ai.h2o.sparkling.ml.models.H2OMOJOSettings",
                                            self.predictionCol,
                                            self.detailedPredictionCol,
                                            self.convertUnknownCategoricalLevelsToNa,
                                            self.convertInvalidNumbersToNa,
                                            self.withContributions,
                                            self.withInternalContributions,
                                            self.withPredictionInterval,
                                            self.withLeafNodeAssignments,
                                            self.withStageResults,
                                            self.dataFrameSerializer,
                                            self.scoringBulkSize)
        return self._java_obj

    @staticmethod
    def default():
        return H2OMOJOSettings()

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOSettings.scala
package ai.h2o.sparkling.ml.models

import ai.h2o.sparkling.ml.params.H2OAlgorithmMOJOParams
import ai.h2o.sparkling.utils.DataFrameSerializer

case class H2OMOJOSettings(
    predictionCol: String = "prediction",
    detailedPredictionCol: String = "detailed_prediction",
    convertUnknownCategoricalLevelsToNa: Boolean = false,
    convertInvalidNumbersToNa: Boolean = false,
    withContributions: Boolean = false,
    withInternalContributions: Boolean = false,
    withPredictionInterval: Boolean = false,
    withLeafNodeAssignments: Boolean = false,
    withStageResults: Boolean = false,
    dataFrameSerializer: String = DataFrameSerializer.default.getClass().getName(),
    scoringBulkSize: Int = 1000)

object H2OMOJOSettings {
  def default = H2OMOJOSettings()

  def createFromModelParams(params: H2OAlgorithmMOJOParams): H2OMOJOSettings = {
    H2OMOJOSettings(
      predictionCol = params.getPredictionCol(),
      detailedPredictionCol = params.getDetailedPredictionCol(),
      withContributions = params.getWithContributions(),
      convertUnknownCategoricalLevelsToNa = params.getConvertUnknownCategoricalLevelsToNa(),
      convertInvalidNumbersToNa = params.getConvertInvalidNumbersToNa(),
      withLeafNodeAssignments = params.getWithLeafNodeAssignments(),
      withStageResults = params.getWithStageResults(),
      dataFrameSerializer = params.getDataFrameSerializer())
  }
}
